paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java

```diff
@@ -49,7 +49,12 @@ public interface Catalog extends AutoCloseable {
      * Get lock factory from catalog. Lock is used to support multiple concurrent writes on the
      * object store.
      */
-    Optional<CatalogLock.Factory> lockFactory();
+    Optional<CatalogLock.LockFactory> lockFactory();
+
+    /** Get lock context for lock factory to create a lock. */
+    default Optional<CatalogLock.LockContext> lockContext() {
```

Comment:
Why do we need to introduce `LockContext`? `Catalog` already provides the `lockFactory` interface. Even if some configuration are needed, they can be obtained in the `options` of the `Catalog`, just like `HiveCatalog`.

Reply:
I'd like to make catalog lock factory(`LockFactory`) extend `Factory` which can be discovered and initialized by `FactoryUtil`, this requires the implementations of `LockFactory` to have an empty constructor. Currently `HiveCatalogLockLockFactory` has parameters such as `hiveConf` and `clientClassName` in its constructor, we need to put them in `LockContext`